power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -45,8 +47,122 @@ struct DatasetInfo {
     Idx batch_size; // for single dataset, the batch size is one
     MetaDataset const* dataset;
     std::vector<ComponentInfo> component_info;
+
+    DatasetInfo(bool is_batch, Idx batch_size, MetaDataset const* dataset)
+        : is_batch(is_batch), batch_size(batch_size), dataset(dataset) {}
 };
 
+template <typename T, dataset_type_tag dataset_type> class ColumnarAttributeRange {
+  public:
+    using Data = std::conditional_t<is_data_mutable_v<dataset_type>, void, void const>;
+
+    class iterator;
+
+    class Proxy {
+      public:
+        using value_type = T;
```

Comment:
The value type should also be a non-const. Return a const value is very tricky. So static assert here that `T` is not const.

Reply:
added `remove_const_t`